examples/systemcalls/systemcalls.c

```diff
@@ -58,7 +80,46 @@ bool do_exec(int count, ...)
  *   as second argument to the execv() command.
  *
 */
-
+// Reference taken from Linux System Programming Textbook
+	pid_t pid = fork();//Create child process
+	//printf("ForkPID:%d\n",pid);
+	int wait_status;
+	int ret_exec = 0;
+	if(pid == -1)
+	{
+	 printf("Child process is not created sucessfully");
+	 return false;
+	}
+	else if(!pid)
+	{
+	 //Ran by child process
+	 ret_exec = execv(command[0],command);
+	 if(ret_exec == -1)
+	 {
+	  perror("Execv failed");
+	  //printf("Execv failed\n");
+     	  exit(EXIT_FAILURE);
+	 } 
+	}
+	else
+	{
+	 //printf("BEPIDis: %d\n",pid);
+	 pid = wait(&wait_status);
+	 //printf("AFPIDis: %d\n",pid);
+        if (WIFEXITED(wait_status) == 1)//Check if exit call is called by the child process
+        {
```

Comment:
@SrirajVemparala watch the code formatting and indents here, suggest either 2 or 4 spaces per indent, be consistent.